packages/status-core/src/utils/public-key-to-color-hash.ts

```diff
@@ -0,0 +1,91 @@
+// TODO?: import only used; or just '@noble/secp256k1' instead
+import * as secp256k1 from 'ethereum-cryptography/secp256k1'
+
+type ColorHash = number[][]
+
+const COLOR_HASH_COLORS_COUNT = 32
+const COLOR_HASH_SEGMENT_MAX_LENGTH = 5
+
+export function publicKeyToColorHash(publicKey: string): ColorHash | undefined {
+  const publicKeyHex = publicKey.replace(/^0[xX]/, '') // ensures hexadecimal digits without "base prefix"
+
+  let compressedPublicKeyDigits: string
+  try {
+    // TODO?: consider obtaining without a crypto lib
+    compressedPublicKeyDigits =
+      secp256k1.Point.fromHex(publicKeyHex).toHex(true) // validates and adds "sign prefix" too
+  } catch (error) {
+    return undefined
+  }
+
+  const colorHashHex = compressedPublicKeyDigits.slice(43, 63)
+  const colorHash = hexToColorHash(
+    colorHashHex,
+    COLOR_HASH_COLORS_COUNT,
+    COLOR_HASH_SEGMENT_MAX_LENGTH
+  )
+
+  return colorHash
+}
+
+export function hexToColorHash(
+  hex: string,
+  colorsCount: number,
+  unitsCount: number
+): ColorHash {
+  const colorIndices = numberToIndices(
+    BigInt(`0x${hex}`),
+    BigInt(colorsCount * unitsCount)
+  )
+  const colorHash = colorIndicesToColorHash(colorIndices, colorsCount)
+
+  return colorHash
+}
+
+// TODO: write as recursive func
+function numberToIndices(number: bigint, base: bigint): bigint[] {
+  const indices: bigint[] = []
+  let nextNumber = number
+
+  if (nextNumber === 0n) {
+    return [0n]
+  }
+
+  while (nextNumber > 0n) {
+    // TODO?: use % instead
+    const modulo = secp256k1.utils.mod(nextNumber, base)
+    nextNumber = nextNumber / base // truncates fractional results
+
+    // TODO?: convert to Number here already
+    indices.push(modulo)
```

Comment:
Non-blocking: Converting to number here wouldn't require use `Number()` in `colorIndicesToColorHash`.